rerun_cpp/src/rerun/sdk_info.hpp

```diff
@@ -1,7 +1,14 @@
 // General information about the SDK.
 #pragma once
 
+#include "error.hpp"
+
 namespace rerun {
     /// The Rerun C++ SDK version as a human-readable string.
     const char* version_string();
+
+    /// Checks whether the version reported by the rerun_c binary matches `sdk_version_string`.
+    ///
+    /// This method is called on various C++ API entry points, calling `Error::handle` on the return value.
+    Error check_binary_and_header_version_match();
```

Comment:
We should make it explicit that this is an internal API thing - users are not expected to need to call this

Reply:
Is it really though? Advanced users might want to call this in their code ahead of time.